File: src/main/Cwe434FileUploadServlet.scala
class Cwe434FileUploadServlet extends Nothing {

  val UPLOAD_DIRECTORY_STRING = "/tmp/uploads"

  @throws[ServletException]
  @throws[IOException]
  protected def doPost(request: Nothing, response: Nothing): Unit = {
    response.setContentType("text/html")
    val out = response.getWriter
    val contentType = request.getContentType
    // the starting position of the boundary header
    val ind = contentType.indexOf("boundary=")
    val boundary = contentType.substring(ind + 9)
    var pLine = new Nothing
    val uploadLocation = new Nothing(UPLOAD_DIRECTORY_STRING) //Constant value
    // verify that content type is multipart form data
    if (contentType != null && (contentType.indexOf("multipart/form-data") ne -1)) { // extract the filename from the Http header
      val br = new Nothing(new Nothing(request.getInputStream))
      pLine = br.readLine
      val filename = pLine.substring(pLine.lastIndexOf("\\"), pLine.lastIndexOf("\""))
      // output the file to the local upload directory
      try {
        val bw = new Nothing(new Nothing(uploadLocation + filename, true))
        var line = null
        while ( {
          (line = br.readLine) != null
        }) {
          if (line.indexOf(boundary) eq -1) {
            bw.write(line)
            bw.newLine
            bw.flush
          }
          //end of for loop}
          bw.close
        }
        catch {
          case ex: Nothing =>

        }
        // output successful upload response HTML page
      }
      else
      { // output unsuccessful upload response HTML page
        return
      }
    }
  }
}

File: src/main/java/Cwe434FileUploadServlet.java
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.*;

public class Cwe434FileUploadServlet extends HttpServlet {


    private static final String UPLOAD_DIRECTORY_STRING = "/tmp";

    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        String contentType = request.getContentType();

// the starting position of the boundary header
        int ind = contentType.indexOf("boundary=");
        String boundary = contentType.substring(ind+9);

        String pLine = new String();
        String uploadLocation = new String(UPLOAD_DIRECTORY_STRING); //Constant value

// verify that content type is multipart form data
        if (contentType != null && contentType.indexOf("multipart/form-data") != -1) {
// extract the filename from the Http header
            BufferedReader br = new BufferedReader(new InputStreamReader(request.getInputStream()));

            pLine = br.readLine();
            String filename = pLine.substring(pLine.lastIndexOf("\\"), pLine.lastIndexOf("\""));


// output the file to the local upload directory
            try {
                BufferedWriter bw = new BufferedWriter(new FileWriter(uploadLocation+filename, true));
                for (String line; (line=br.readLine())!=null; ) {
                    if (line.indexOf(boundary) == -1) {
                        bw.write(line);
                        bw.newLine();
                        bw.flush();
                    }
                } //end of for loop
                bw.close();


            } catch (IOException ex) {
                return;
            }
// output successful upload response HTML page
        }
// output unsuccessful upload response HTML page
        else
        { return; }
    }

}
